std/algorithm/mutation.d

```diff
@@ -891,6 +891,7 @@ unittest
 
     move(obj2, obj3);
     assert(obj3 is obj1);
+    assert(obj2 is obj1);
```

Comment:
See here: https://github.com/ntrel/phobos/blob/move-docs/std/algorithm/mutation.d#L1206

`move` forwards to `moveEmplace` for the real work. Notice `source` only gets cleared if it's a struct with an elaborate destructor or elaborate postblit, otherwise source is left unchanged. I think the docs could make this clearer. The user needs to understand that a copy is made unless this would cause problems, and only then is a true move made. That's why I added the assert.

(Personally I would prefer a true move is always made to avoid holding on to GC memory accidentally).